contracts/SoulBoundToken/BasicSBT.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+// SpartanLabs Contracts (SBT)
+pragma solidity ^0.8.0;
+
+import "./utils/Ownable.sol";
+
+/**
+ * @dev Implementation of Soul Bound Token (SBT)
+ * following Vitalik's co-authored whitepaper at: https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4105763
+ *
+ * Contract provides a basic Soul Bound Token mechanism, where address can mint SBT with their private data.
+ *
+ * By default, the owner account will be the one that deploys the contract. This
+ * can later be changed with {transferOwnership}.
+ */
+contract BasicSBT is Ownable {
+    // Name for the SBT
+    string public _name;
+
+    // Symbol for the SBT
+    string public _symbol;
+
+    // Total count of SBT
+    uint256 public _totalSBT; // TODO: Should we keep track of how many SBT?
+
+    // Mapping from SBT ID to owner address
+    mapping(uint256 => address) private _owners; // TODO : Do we even need to keep track of owners?
+
+    // Mapping between address and the soul
+    mapping(address => Soul) private souls;
+
+    /**
+     * @dev Struct `Soul` contains the soulbound token information for a given address.
+     * The fields within the struct can be be edited for the usecase of the SBT.
+     *
+     * The fields of `identity` and `url` can be hashed for privacy.
+     */
+    struct Soul {
+        string identity;
+        string url;
+    }
+
+    // Events
+    event Mint(address _soul);
+    event Burn(address _soul);
+    event Update(address _soul);
+
+    /**
+     * @dev This modifier checks that the address passed in is not the zero address.
+     */
+    modifier validAddress(address _addr) {
+        require(_addr != address(0), "Not valid address");
+        _;
+    }
+
+    /**
+     * @dev Initializes the contract by setting a `name` and a `symbol` to the SBT
+     */
+    constructor(string memory name_, string memory symbol_) {
+        _name = name_;
+        _symbol = symbol_;
+        _totalSBT = 0;
+    }
+
+    /**
+     * @dev Mints `SBT` and transfers it to `_soul`.
+     *
+     * Emits a {Mint} event.
+     */
+    function mint(address _soul, Soul memory _soulData)
+        external
+        virtual
+        validAddress(_soul)
+    {
+        require(!hasSoul(_soul), "Soul already exists");
+        souls[_soul] = _soulData;
+        _totalSBT++;
+        _owners[_totalSBT] = _soul;
+        emit Mint(_soul);
+    }
+
+    /**
+     * @dev Destroys SBT for a given address.
+     *
+     * Requirements:
+     * Only the owner of the SBT can destroy it.
+     * Emits a {Burn} event.
+     */
+    function burn(address _soul) external virtual validAddress(_soul) {
```

Comment:
This seems iffy - this means that I can burn any soul where the data no longer conforms to what I'd want it to be - for e.g. if I don't like my credit score I'm gonna burn it 😅

Reply:
hmm that is true, people can easily restart their SBT. Don't think we would want that. 

Maybe we can make it such that only owners can burn it?

Reply:
Maybe can do similar to the data part? That they can propose but ultimately the burning is by the owners? Anyway if the owners data is not accurate - people won't read their data - social reputation

Reply:
that makes sense!